src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -63,14 +72,46 @@ def _process_model_before_weight_loading(self, model, **kwargs):
         from compressed_tensors.quantization import apply_quantization_config
 
         ct_quantization_config = self.compressor.quantization_config
-        apply_quantization_config(model, ct_quantization_config, run_compressed=True)
+
+        if self.run_compressed and self.is_compressed:
+            apply_quantization_config(model, ct_quantization_config, run_compressed=True)
 
     def _process_model_after_weight_loading(self, model, **kwargs):
-        pass
+        """Decompress loaded model if necessary - need for qat"""
+
+        if not self.run_compressed or not self.is_compressed:
+            config = kwargs.get("config", None)
+            cache_path = config._name_or_path
+            if not os.path.exists(cache_path):
+                from huggingface_hub import hf_hub_download
+
+                from transformers import TRANSFORMERS_CACHE
+                from transformers.utils import http_user_agent
+
+                user_agent = http_user_agent()
+                config_file_path = hf_hub_download(
+                    repo_id=cache_path,
+                    filename="config.json",
+                    cache_dir=TRANSFORMERS_CACHE,
+                    force_download=False,
+                    user_agent=user_agent,
+                )
+                cache_path = os.path.sep.join(config_file_path.split(os.path.sep)[:-1])
+
+            from compressed_tensors.quantization import QuantizationStatus
+
+            self.compressor.quantization_config.quantization_status = QuantizationStatus.FROZEN
+
+            self.compressor.decompress(model_path=cache_path, model=model)
```

Comment:
If I understood correctly, you are trying to get the config file of the original model. Could you tell me why this is needed to decompress the model ?

Reply:
The only thing I need here is the path to the model safetensors. They are in a compressed format so by knowing the path I can decompress them

Reply:
If there is a better way to get the download folder path, lemme know

Reply:
config._name_or_path won't give you necessarily the path to the quantized model as you can see here:  https://huggingface.co/neuralmagic/Meta-Llama-3-8B-Instruct-FP8/blob/main/config.json#:~:text=%22_name_or_path%22%3A%20%22meta%2Dllama/Meta%2DLlama%2D3%2D8B%2DInstruct%22%2C

Reply:
@SunMarc 
`self.compressor.decompress(model_path=cache_path, model=model)`

:param model_path: path to compressed weights
:param model: pytorch model to load decompressed weights into